prow/config/config_test.go

```diff
@@ -205,6 +205,14 @@ func TestTideMergeMethod(t *testing.T) {
 	}
 }
 
+func TestTideQueries(t *testing.T) {
+	for i, tq := range c.Tide.Queries {
+		if err := tq.Validate(); err != nil {
```

Comment:
We should be running this as part of prod code so all deployments can take advantage of it.

Reply:
we should also more broadly make these config tests properly generic and split out any bits specific to the kubernetes/* deployment to something else.

Reply:
+1 

Do broadly useful tests on config load, do k8s-specific in out-of-band test

Reply:
Done.